src/XR/features/WebXREyeTracking.ts

```diff
@@ -0,0 +1,139 @@
+    /**
+     * This observable will notify registered observers when eye tracking ends
+     */
+    public readonly onEyeTrackingEndedObservable: Observable<void> = new Observable();
+    /**
+     * This observable will notify registered observers on each frame that has valid tracking
+     */
+    public readonly onEyeTrackingFrameUpdateObservable: Observable<Ray> = new Observable();
+
+    /**
+     * Creates a new instance of the XR eye tracking feature.
+     * @param _xrSessionManager An instance of WebXRSessionManager.
+     */
+    constructor(_xrSessionManager: WebXRSessionManager) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "eye-tracking";
+        if (this._xrSessionManager.session) {
+            this._init();
+        } else {
+            this._xrSessionManager.onXRSessionInit.addOnce(() => {
+                this._init();
+            });
+        }
+    }
+
+    /**
+     * Dispose this feature and all of the resources attached.
+     */
+    public dispose(): void {
+        super.dispose();
+
+        this._xrSessionManager.session.removeEventListener("eyetrackingstart", this._eyeTrackingStartListener);
+        this._xrSessionManager.session.removeEventListener("eyetrackingend", this._eyeTrackingEndListener);
+
+        this.onEyeTrackingStartedObservable.clear();
+        this.onEyeTrackingEndedObservable.clear();
+        this.onEyeTrackingFrameUpdateObservable.clear();
+    }
+
+    /**
+     * Returns whether the gaze data is valid or not
+     * @returns true if the data is valid
+     */
+    public get isEyeGazeValid(): boolean {
+        return !!this._gazeRay;
+    }
+
+    /**
+     * Get a reference to the gaze ray. This data is valid while eye tracking persists, and will be set to null when gaze data is no longer available
+     * @returns a reference to the gaze ray if it exists and is valid, returns null otherwise.
+     */
+    public getEyeGaze(): Nullable<Ray> {
+        return this._gazeRay;
+    }
+
+    protected _onXRFrame(frame: XRFrame) {
+        if (!this.attached || !frame) {
+            return;
+        }
+
+        if (this._latestEyeSpace && this._gazeRay) {
+            let pose = frame.getPose(this._latestEyeSpace, this._xrSessionManager.referenceSpace);
+            if (pose) {
+                this._gazeRay.origin.set(pose.transform.position.x, pose.transform.position.y, pose.transform.position.z);
+                const quat = pose.transform.orientation;
+                TmpVectors.Quaternion[0].set(quat.x, quat.y, quat.z, quat.w);
+
+                if (!this._xrSessionManager.scene.useRightHandedSystem) {
+                    this._gazeRay.origin.z *= -1;
+                    TmpVectors.Quaternion[0].z *= -1;
+                    TmpVectors.Quaternion[0].w *= -1;
+                }
+
+                Vector3.Forward().rotateByQuaternionToRef(TmpVectors.Quaternion[0], this._gazeRay.direction);

```

Comment:
@rickfromwork , could we change this one before the merge ? Vector3.Forward() to smthg like Vector3.ForwardReadOnly